runtime/src/main/java/org/corfudb/runtime/clients/LayoutHandler.java

```diff
@@ -94,4 +105,116 @@ private static Object handleAlreadyBootstrap(CorfuMsg msg,
             throws Exception {
         throw new AlreadyBootstrappedException();
     }
+
+    // Protobuf region
+
+    /**
+     * Handle a layout response from the server.
+     *
+     * @param msg The layout response message.
+     * @param ctx The context the message was sent under.
+     * @param r A reference to the router.
+     * @return {@link Layout} sent back from server.
+     */
+    @ResponseHandler(type = PayloadCase.LAYOUT_RESPONSE)
+    private static Object handleLayoutResponse(ResponseMsg msg, ChannelHandlerContext ctx,
+                                               IClientRouter r) {
+        LayoutResponseMsg layoutResponse = msg.getPayload().getLayoutResponse();
+        RpcCommon.LayoutMsg layoutMsg = layoutResponse.getLayout();
+
+        return CorfuProtocolCommon.getLayout(layoutMsg);
+    }
+
+    /**
+     * Handle a prepare layout response from the server.
+     *
+     * @param msg The prepare layout response message.
+     * @param ctx The context the message was sent under.
+     * @param r A reference to the router.
+     * @return {@link LayoutPrepareResponse} if ACK, throw an {@link OutrankedException} if REJECT.
+     */
+    @ResponseHandler(type = PayloadCase.PREPARE_LAYOUT_RESPONSE)
+    private static Object handlePrepareLayoutResponse(ResponseMsg msg, ChannelHandlerContext ctx,
+                                                      IClientRouter r) {
+        PrepareLayoutResponseMsg prepareLayoutMsg = msg.getPayload().getPrepareLayoutResponse();
+        PrepareLayoutResponseMsg.Type type = prepareLayoutMsg.getRespType();
+        long rank = prepareLayoutMsg.getRank();
+        Layout layout = CorfuProtocolCommon.getLayout(prepareLayoutMsg.getLayout());
+
+        switch (type) {
+            case ACK: return new LayoutPrepareResponse(rank, layout);
```

Comment:
`LayoutPrepareResponse` no longer needs to implement `ICorfuPayoad`. It can be removed, along with the related serialization methods.